core/extension/package.json

```diff
@@ -16,7 +16,8 @@
     "cover": "",
     "docs": "",
     "lint": "",
-    "test": ""
+    "test": "",
+    "generate": "node ./codeGen/generate-exports.js '[\"@itwin/core-frontend\",\"core/frontend\"] [\"@itwin/core-common\",\"core/common\"]'"
```

Comment:
What is the workflow here for someone making an update to the Extension API that we're expecting or in the normal dev workflow?

With the current setup, I'd assume:

1. `rush build`
2. `rush extract-api`
3. `cd ./core/extension`
4. `npm run generate`

If so, how do we envision enforcing that someone does this if they change an API?

There are 4 main workflows we need to work with:

1. A new `public` API is added
    - This forces extract-api to be run and will catch any accidental mis-match. Forces a review of the api.md file. Has nothing to do with Extensions
2. The `extensions` tag is added to an API which is not `public`
    - Extract API CI job will catch the mis-match and fail the build
3. The `extensions` tag is added to a `public` API
    - This will pass the extract-api CI job but will not do anything to force the author to run the generate command. How do we intend on reviewing the addition to the Extension API?
4. The `extensions` tag is accidentally removed from a `public` API
    - I think this should be caught by peer review but it will never show up as a change to either `./core/frontend/extension/ExtensionRuntime.ts` or `./core/extension/*_types.d.ts`

There are some other various permutations on the ones above but I think those cover the main cases.

Reply:
let's discuss this on Friday. I feel like 3 and 4 could/should be automated